satellite/audit/reservoirservice.go

```diff
@@ -46,21 +52,95 @@ func (service *ReservoirService) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	service.log.Info("audit 2.0 is starting up")
 
+	var group errgroup.Group
+	group.Go(func() error {
+		return service.populateQueueJob(ctx)
+	})
+
+	for i := 0; i < service.config.WorkerCount; i++ {
+		group.Go(func() error {
+			return service.worker(ctx)
+		})
+	}
+
+	return group.Wait()
+}
+
+func (service *ReservoirService) populateQueueJob(ctx context.Context) error {
 	return service.Loop.Run(ctx, func(ctx context.Context) (err error) {
 		defer mon.Task()(&ctx)(&err)
-		pathCollector := NewPathCollector(service.reservoirSlots, service.rand)
+
+		select {
+		case <-service.closed:
+			return nil
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		pathCollector := NewPathCollector(service.config.Slots, service.rand)
 		err = service.MetainfoLoop.Join(ctx, pathCollector)
 		if err != nil {
 			service.log.Error("error joining metainfoloop", zap.Error(err))
 			return nil
 		}
-		service.Reservoirs = pathCollector.Reservoirs
+
+		var queue []storj.Path
+		queuePaths := make(map[storj.Path]bool)
```

Comment:
nit: I would use `map[storj.Path]struct{}` to express that the value is meaningless